config/v1/types_dns.go

```diff
@@ -26,6 +26,25 @@ type DNSSpec struct {
 	// For example, given the base domain `openshift.example.com`, an API server
 	// DNS record may be created for `cluster-api.openshift.example.com`.
 	BaseDomain string `json:"baseDomain"`
+	// publicZone is the location where all the DNS records that are publicly accessible to
+	// the internet exist.
+	// If this field is nil, no public records should be created.
+	// +optional
+	PublicZone *DNSZone `json:"publicZone,omitempty"`
```

Comment:
Is this allowed to be changed?

Reply:
Technically no.

Reply:
> Technically no.

Why not?  "I don't want public DNS anymore" and "actually, I want it back" seem like reasonable admin decisions, right?

Reply:
> > Technically no.
> 
> Why not? "I don't want public DNS anymore" and "actually, I want it back" seem like reasonable admin decisions, right?

Oh Yeah, true